src/PaprController.sol

```diff
@@ -492,6 +500,32 @@ contract PaprController is
         emit IncreaseDebt(account, asset, amount);
     }
 
+    /// @notice caches and returns the minimum of the passed price and the max price as well as the timestamp
+    /// @dev max price computed by MAX_PER_SECOND_PRICE_GROWTH * time elapsed since the cache was last updated
+    /// @dev time elapsed maxes at 2 days such that price can never grow by more than 100% between two successive
+    ///      increase debt events for the same asset
+    function _cacheAndReturnPriceOrMaxPrice(ERC721 asset, uint256 price) internal returns (uint256) {
```

Comment:
just a small design question, isn't this more particular to the underwriting engine and would make more sense to put in ReservoirOracleUnderwriter.sol?

just thinking that this change is only particular to us having a single point of failure in our oracle underwriting, but could imagine in the future if we had multiple oracles, we wouldn't need this at all

just a thought, but don't feel too strongly about it

Reply:
ended up doing this